zigzag/stages/SpatialMappingGeneratorStage.py

```diff
@@ -170,6 +170,19 @@ def limit_unrolling_to_mem_bandwidth(
         self, mapping: dict[OADimension, dict[LayerDim, UnrollFactorInt]]
     ) -> dict[OADimension, dict[LayerDim, UnrollFactorInt]]:
         """! Scale the given unroll factors such that they do not exceed the bandwidths of the memory structure"""
+
+        def conditional_log(layer_dim: LayerDim, oa_dim: OADimension, value: int, mem_name: str):
+            # Don't log if user has defined an unrolling for a different layer dim
```

Comment:
The question here is exactly what the condition below states, right?
`do_not_log = oa_dim in self.provided_mapping and layer_dim not in self.provided_mapping[oa_dim]`

In my opinion, this declaration is expressive enough and shouldn't be replicated into a comment

Reply:
I understand. I agree the condition has been clearly expressed in the code. But the comment here is re-declaring but only partially. It would be better to describe the entire declaration but still keep it concise as it was.